ooni/tests/test_scheduler.py

```diff
@@ -51,8 +52,40 @@ def task(subself):
             self.assertEqual(len(in_file.readlines()), 1)
 
         self.assertEqual(dummy_st.should_run, False)
+        self.assertFalse(os.path.islink(os.path.join(scheduler_directory, identifier + '.lock')))
         shutil.rmtree(scheduler_directory)
 
+    def test_thundering_herd(self):
+        lockno = int(os.getenv('TTH_LOCKNO', str(sys.getrecursionlimit() + 16)))
+        scheduler_directory = tempfile.mkdtemp()
+        counter = os.path.join(scheduler_directory, 'counter')
+        class DummyST(ScheduledTask):
+            @defer.inlineCallbacks
+            def task(subself):
+                self.assertTrue(os.path.islink(os.path.join(scheduler_directory, subself.identifier + '.lock')))
+                with open(counter, 'w+') as fd:
+                    data = fd.read()
+                    fd.seek(0)
+                    if not data:
+                        fd.write('1')
+                        yield task.deferLater(reactor, 1, lambda: 42)
+                    else:
+                        # should not write anything as followup tasks should not be run
+                        fd.write(str(int(data) + 1))
+                        yield 34
```

Comment:
This generator should be yielding a deferred and not an int. Otherwise it will trigger some exception.